scenarios/csharp/dotnetcore/post-call-analytics/Program.cs

```diff
@@ -0,0 +1,114 @@
+            {
+                await audioStream.CopyToAsync(memoryStream);
+                byteArray = memoryStream.ToArray();
+            }
+            return new ByteArrayContent(byteArray);
+        }
+
+        internal static async Task<string?> TranscribeAsync(string speechKey, string speechRegion, FileInfo inputAudio)
+        {
+            var speechEndpoint = $"https://{speechRegion}.api.cognitive.microsoft.com/speechtotext/transcriptions:transcribe?api-version=2024-05-15-preview";
+
+            var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", speechKey);
+
+            var multipartFormDataContent = new MultipartFormDataContent();
+            var fastTranscriptionOptions = new FastTranscriptionOptions();
+            var fastTranscriptionOptionsDefinition = JsonContent.Create(fastTranscriptionOptions);
+            multipartFormDataContent.Add(fastTranscriptionOptionsDefinition, "definition");
+
+            var streamContent = await GetAudioContentAsync(inputAudio);
+            streamContent.Headers.Add("Content-Type", "multipart/form-data");
+
+            // Speech Fast TranscribeAsync requires a filename to be specified when sending a file.
+            multipartFormDataContent.Add(streamContent, "audio", "audio.wav");
+
+            var response = await httpClient.PostAsync(speechEndpoint, multipartFormDataContent, CancellationToken.None);
+            Console.WriteLine($"{speechEndpoint} : {response.StatusCode}");
+
+            var content = await response.Content.ReadAsStringAsync();
+            var json = JsonDocument.Parse(content).RootElement;
+
+            var combinedTranscript = json.GetProperty("combinedPhrases")[0].GetProperty("text").GetString();
+            return combinedTranscript;
+        }
+
+        internal static async Task<string> SummarizeAsync(string openAiKey, string openAiEndpoint, string deploymentOrModelName, string transcription)
+        {
+            var azureClient = new AzureOpenAIClient(new Uri(openAiEndpoint), new ApiKeyCredential(openAiKey));
+            var chatClient = azureClient.GetChatClient(deploymentOrModelName);
+
+            var prompt = $"You are an AI assistant that helps extract information from customer call center transcripts. " +
+                $"Summarize the conversation in a couple sentences. \n\"\"\"\n{transcription}\n\"\"\"\n";
+
+            var completion = await chatClient.CompleteChatAsync(
+                new SystemChatMessage(prompt)
+            );
+
+            Console.WriteLine($"{openAiEndpoint} : {completion.GetRawResponse().Status}");
+
+            var summary = completion.Value.Content[0].Text;
+            return summary;
+        }
+
+        internal static async Task AnalyzeAudioAsync(string speechKey, string speechRegion, FileInfo? inputAudio, string openAiKey, string openAiEndpoint, string deploymentOrModelName)
```

Comment:
Why support a nullable FileInfo, and then error if it's null?

Reply:
Removed ? and added a check for inputAudio.Exists